src/QMCWaveFunctions/tests/test_ConstantSPOSet.cpp

```diff
@@ -0,0 +1,193 @@
+  using GradMatrix = OrbitalSetTraits<ValueType>::GradMatrix;
+
+  ValueVector row0{ValueType(0.92387953), ValueType(0.92387953)};
+  ValueVector row1{ValueType(0.29131988), ValueType(0.81078057)};
+
+  GradVector  grow0{GradType({ -2.22222, -1.11111, 0.33333}), GradType({8.795388,-0.816057,-0.9238793})};
+  GradVector  grow1{GradType({  2.22222,  1.11111, -0.33333}), GradType({-8.795388,0.816057,0.9238793})};
+
+  ValueVector lrow0{ValueType(-0.2234545), ValueType(0.72340234)};
+  ValueVector lrow1{ValueType(-12.291810), ValueType(6.879057)};
+
+  
+  ValueMatrix spomat;
+  GradMatrix gradspomat;
+  ValueMatrix laplspomat;
+
+  spomat.resize(nelec, norb);
+  gradspomat.resize(nelec, norb);
+  laplspomat.resize(nelec, norb);
+
+  for (int iorb = 0; iorb < norb; iorb++)
+  {
+    spomat(0, iorb) = row0[iorb];
+    spomat(1, iorb) = row1[iorb];
+    
+    gradspomat(0,iorb) = grow0[iorb];
+    gradspomat(1,iorb) = grow1[iorb];
+   
+    laplspomat(0,iorb) = lrow0[iorb];
+    laplspomat(1,iorb) = lrow1[iorb];
+  }
+
+
+  const SimulationCell simulation_cell;
+  ParticleSet elec(simulation_cell);
+
+  elec.create({3});
+
+  ValueVector psiV = {0.0, 0.0};
+  ValueVector psiL = {0.0, 0.0};
+  GradVector psiG;
+  psiG.resize(norb);
+
+  //Test of value only constructor.
+  auto sposet = std::make_unique<ConstantSPOSet>("constant_spo", spomat);
+  sposet->evaluateValue(elec, 0, psiV);
+
+  CHECK(psiV[0] == row0[0]);
+  CHECK(psiV[1] == row0[1]);
+
+
+  psiV = 0.0;
+
+  sposet->evaluateValue(elec, 1, psiV);
+  CHECK(psiV[0] == row1[0]);
+  CHECK(psiV[1] == row1[1]);
+
+  psiV = 0.0;
+
+  sposet->evaluateVGL(elec, 1, psiV, psiG, psiL);
+  CHECK(psiV[0] == row1[0]);
+  CHECK(psiV[1] == row1[1]);
+
+  CHECK(psiG[0][0] == ValueType(0.0));
+  CHECK(psiG[0][1] == ValueType(0.0));
+  CHECK(psiG[0][2] == ValueType(0.0));
+
+  CHECK(psiG[1][0] == ValueType(0.0));
+  CHECK(psiG[1][1] == ValueType(0.0));
+  CHECK(psiG[1][2] == ValueType(0.0));
+
+  CHECK(psiL[0] == ValueType(0.0));
+  CHECK(psiL[1] == ValueType(0.0));
+
+  //Test of value and gradient constructor.
+  auto sposet_vg = std::make_unique<ConstantSPOSet>("constant_spo", spomat, gradspomat);
+  psiV=0;
+  psiG=0;
+  psiL=0;
+  sposet_vg->evaluateVGL(elec,1,psiV,psiG,psiL);
+   
+  CHECK(psiV[0] == row1[0]);
+  CHECK(psiV[1] == row1[1]);
+  
+  CHECK(psiG[0][0] == grow1[0][0]); 
+  CHECK(psiG[0][1] == grow1[0][1]); 
+  CHECK(psiG[0][2] == grow1[0][2]); 
+
+  CHECK(psiG[1][0] == grow1[1][0]); 
+  CHECK(psiG[1][1] == grow1[1][1]); 
+  CHECK(psiG[1][2] == grow1[1][2]); 
+
+  CHECK(psiL[0] == ValueType(0.0));
+  CHECK(psiL[1] == ValueType(0.0));
+
+  //Test of value, gradient, and laplacian constructor.
+  auto sposet_vgl = std::make_unique<ConstantSPOSet>("constant_spo", spomat, gradspomat,laplspomat);
+  psiV=0;
+  psiG=0;
+  psiL=0;
+  sposet_vgl->evaluateVGL(elec,1,psiV,psiG,psiL);
+   
+  CHECK(psiV[0] == row1[0]);
+  CHECK(psiV[1] == row1[1]);
+  
+  CHECK(psiG[0][0] == grow1[0][0]); 
```

Comment:
Use `src/Utils/forTesting/checkMatrix.hpp`?  This is painful to look at and it has pretty decent reporting if there is an bad element.

Reply:
Also exact tool I need.  Thank you.  Will fix.

Reply:
Do we have anything like checkMatrix for vectors?

Reply:
ParticleGradient and GradTypes are vectors of vectors, so for now I'm just checking them with loops.  Need to write some checker functions to handle vectors and vectors of vectors.